fission/src/systems/scene/GizmoSceneObject.ts

```diff
@@ -0,0 +1,194 @@
+        this._gizmo = new TransformControls(World.SceneRenderer.mainCamera, World.SceneRenderer.renderer.domElement)
+        this._gizmo.setMode(mode)
+
+        World.SceneRenderer.RegisterGizmoSceneObject(this)
+    }
+
+    public Setup(): void {
+        // adding the mesh and gizmo to the scene
+        World.SceneRenderer.AddObject(this._obj)
+        World.SceneRenderer.AddObject(this._gizmo)
+
+        // forcing the gizmo to rotate and transform with the object
+        this._gizmo.setSpace("local")
+        this._gizmo.attach(this._obj)
+
+        this._gizmo.addEventListener("dragging-changed", (event: { target: TransformControls; value: unknown }) => {
+            // disable orbit controls when dragging the transform gizmo
+            const gizmoDragging = World.SceneRenderer.IsAnyGizmoDragging()
+            if (!event.value && !gizmoDragging) {
+                World.SceneRenderer.orbitControls.enabled = true // enable orbit controls when not dragging another transform gizmo
+            } else if (!event.value && gizmoDragging) {
+                World.SceneRenderer.orbitControls.enabled = false // disable orbit controls when dragging another transform gizmo
+            } else {
+                World.SceneRenderer.orbitControls.enabled = !event.value // disable orbit controls when dragging transform gizmo
+            }
+
+            if (event.target.mode === "translate") {
+                // disable other gizmos when translating
+                Array.from(World.SceneRenderer.sceneObjects.values())
+                    .filter(obj => obj instanceof GizmoSceneObject)
+                    .forEach(obj => {
+                        if (obj.gizmo.object === event.target.object && obj.gizmo.mode !== "translate") {
+                            obj.gizmo.dragging = false
+                            obj.gizmo.enabled = !event.value
+                            return
+                        }
+                    })
+            } else if (
+                event.target.mode === "scale" &&
+                (InputSystem.isKeyPressed("ShiftRight") || InputSystem.isKeyPressed("ShiftLeft"))
+            ) {
+                // scale uniformly if shift is pressed
+                event.target.axis = "XYZE"
+            } else if (event.target.mode === "rotate") {
+                // scale on all axes
+                Array.from(World.SceneRenderer.sceneObjects.values())
+                    .filter(obj => obj instanceof GizmoSceneObject)
+                    .forEach(obj => {
+                        if (
+                            obj.gizmo.mode === "scale" &&
+                            event.target !== obj.gizmo &&
+                            obj.gizmo.object === event.target.object
+                        ) {
+                            obj.gizmo.dragging = false
+                            obj.gizmo.enabled = !event.value
+                            return
+                        }
+                    })
+            }
```

Comment:
Would a switch be better here? I'm not sure if the if will be optimized, especially since you're checking inputs in one of the conditions.